src/Store/Azure/SearchStore.php

```diff
@@ -88,8 +89,8 @@ private function convertArrayToDocument(array $data): Document
     {
         return new Document(
             id: Uuid::fromString($data['id']),
-            text: null,
-            vector: null,
+            text: $data['text'],
+            vector: $data[$this->vectorFieldName] ? new Vector($data[$this->vectorFieldName]) : null,
```

Comment:
hmm I would not go this direction here, I would assume, that `$data[$this->vectorFieldName]` is always available.